packages/react/src/avatar/image/useImageLoadingStatus.ts

```diff
@@ -0,0 +1,43 @@
+'use client';
+import * as React from 'react';
+
+export type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error';
+
+export function useImageLoadingStatus(
+  src?: string,
+  referrerPolicy?: React.HTMLAttributeReferrerPolicy,
+): ImageLoadingStatus {
+  const [loadingStatus, setLoadingStatus] = React.useState<ImageLoadingStatus>('idle');
+
+  React.useLayoutEffect(() => {
```

Comment:
Use `useEnhancedEffect` from our utils instead of `React.useLayoutEffect` to prevent errors during SSR.